solutions/day7.py

```diff
@@ -0,0 +1,68 @@
+
+from utilities import input_reader
+
+def parse_input(input_list: list) -> tuple:
+    left_list, right_list = [], []
+    for line in input_list:
+        split_line = line.split(":")
+        left_list.append(int(split_line[0]))
+        right_list.append(list(int(i) for i in split_line[1].split()))
+    return left_list, right_list
+
+def calculate(nums: list, ops: list):
+    start = nums[0]
+    i = 1
+    while i < len(nums):
+        num, op = nums[i], ops[i-1]
+        if op=="+":
+            start += num
+        elif op=="c":
+            start = int(str(start)+str(num))
+        else:
+            start *= num
+        i += 1
+    
+    return start
+
+def generate_perms(size: int, ops_list: list, cats: bool, ops=[]) -> None:
+
+    if len(ops)==size:
+        ops_list.append(ops)
+        return
+    
+    copy1 = ops.copy()
+    copy1.append("+")
+    copy2 = ops.copy()
+    copy2.append("*")
+    if cats:
+        copy3 = ops.copy()
+        copy3.append("c")
+        generate_perms(size, ops_list, cats, copy3)
+    generate_perms(size, ops_list, cats, copy1)
+    generate_perms(size, ops_list, cats, copy2)
+
+
+def solution_one(left_nums: list, right_nums: list, cats_allowed: bool) -> int:
```

Comment:
You use this function as the solution for part one and for part two.  I recommend renaming it to simply `solution`
```suggestion
def solution(left_nums: list, right_nums: list, cats_allowed: bool) -> int:
```